### PR TITLE
[WIP] coverity fixes

### PR DESCRIPTION
**Pull request by szalpal**
#### Why we need this PR?
*Pick one*
- It fixes a bug *bug description*
- It adds new feature needed because *why we need this feature*
- Refactoring to improve *what*

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
 - What was changed, added, removed?
 - What is most important part that reviewers should focus on?
 - Was this PR tested? How?
 - Were docs and examples updated, if necessary?

**JIRA TASK**: [DALI-XXXX]

### CODE REVIEW

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

File: dali/pipeline/operators/optical_flow/optical_flow.h [hunk 1]
```diff
@@ -65,7 +65,9 @@ class OpticalFlow : public Operator<Backend> {
           optical_flow_(std::unique_ptr<optical_flow::OpticalFlowAdapter<ComputeBackend>>(
                   new optical_flow::OpticalFlowStub<ComputeBackend>(of_params_))),
           image_type_(spec.GetArgument<decltype(this->image_type_)>(detail::kImageTypeArgName)),
-          device_id_(spec.GetArgument<int>("device_id")) {
+          device_id_(spec.GetArgument<int>("device_id")),
+          frames_width_(-1), frames_height_(-1), depth_(-1), nsequences_(-1), hints_width_(-1),
```
Comment (jantonguirao) [inline dali/pipeline/operators/optical_flow/optical_flow.h]: how about doing that at the member variable definition instead?

File: dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api_test.cc [hunk 2]
```diff
@@ -54,7 +54,7 @@ class nvjpegDecodeDecoupledAPITest : public GenericDecoderTest<ImgType> {
   }
 
  private:
-  unsigned int hybrid_huffman_threshold_;
+  unsigned int hybrid_huffman_threshold_ = -1;
```
Comment (klecki) [inline dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api_test.cc]: Isn't it weird to initialize unsigned int to -1?

File: dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_split_test.cc [hunk 3]
```diff
@@ -56,7 +56,7 @@ class ImageDecoderSplitTest_GPU : public GenericDecoderTest<ImgType> {
   }
 
  private:
-  unsigned int hybrid_huffman_threshold_;
+  unsigned int hybrid_huffman_threshold_ = -1;
```
Comment (klecki) [inline dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_split_test.cc]: And here

**Comment by JanuszL** [comment]
!build

File: dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_split_test.cc (hunk 3 above)
Comment (JanuszL) [reply dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_split_test.cc]: Done

File: dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api_test.cc (hunk 2 above)
Comment (JanuszL) [reply dali/pipeline/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api_test.cc]: Done

File: dali/pipeline/operators/optical_flow/optical_flow.h (hunk 1 above)
Comment (JanuszL) [reply dali/pipeline/operators/optical_flow/optical_flow.h]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**